### ISSUE TITLE
Optimize search of roll_intervals?

### ISSUE THREAD

**Issue by jeanconn** [sot/sparkles#148]
For the project to perhaps make a specialized yoshi to find "a" passable star catalog given an allowed roll range for a target, it seems like the sparkles roll optimization is 99% of the way there.  I think it would need:

1) an option to *stop* looping over the intervals after a "passing" catalog as been found.
2) a speed up to try the roll/roll_intervals with the most-likely-to-succeed stars first.  I'm thinking here that in the uniq_ids method, it might not be dreadful to quickly use either magnitude or acq_success_prob on, say, the brightest acq candidates, to pre-sort the roll_intervals before looping over them.

@taldcroft what are your thoughts, if any on ordering/optimizing the roll_intervals for this application (quickest determination of existence of a passing catalog for a target)?

**Comment by taldcroft** [comment]
Idea 1 is easy.  Idea 2 will take some work. Maybe there is a low-hanging idea for ordering based on just mags that can help out.  I.e. compute P2 based on just mags (ignoring the full p_acq that does spoilers etc) for the 5 brightest stars.

Beyond that I would push back on spending time optimizing this code before we actually have a concrete idea for tool requirements and can do profiling to understand what needs to be improved, and if that is even needed.

**Comment by jeanconn** [comment]
Right.  I did "idea 1" without much effort.  Idea 2 seems doable using wether mags directly or the acq_success prob (which is really a curve defined by mag), but the star index handling for the acquisition candidates was a little confusing on my first go.   I was mostly just trying to evaluate level-of-effort for 2 from you because "it seems like a good idea on paper" to me that could have some bang-for-bunch, and this is all your code.
